### PR TITLE
add websocket handshake_auth.go

### PR DESCRIPTION
**Pull request by deepziyu**
add a handshake-plugin of  websocket  to auth by URL.Query info
Usage:
@see ```func TestHandshakeWebsocketAuth```

### CODE REVIEW

**Comment by henrylee2cn** [comment]
Good job, I think it is basically ok, but you need to add some comments

File: mixer/websocket/handshake_auth.go [hunk 1]
```diff
@@ -6,15 +19,21 @@ import (
...
 
+// NewHandshakeAuthPlugin creates a handshake auth plugin for server.
 func NewHandshakeAuthPlugin(ckFn Checker, apFn Acceptor) *HandshakeAuthPlugin {
 	return &HandshakeAuthPlugin{
 		CheckFunc:  ckFn,
 		AcceptFunc: apFn,
 	}
 }
 
+// Checker provide authenticated erpc.Session
```
Comment (henrylee2cn) [inline mixer/websocket/handshake_auth.go:30]: Should be "Acceptor" ?

Comment (deepziyu) [reply mixer/websocket/handshake_auth.go:30]: 是的

**Comment by yangwenmai** [comment]
maybe you can set your github email for github push account.

**Comment by deepziyu** [comment]
> maybe you can set your github email for github push account.

thx for suggest

**Comment by henrylee2cn** [comment]
@deepziyu Are you ready to merge?

**Comment by deepziyu** [comment]
> @deepziyu Are you ready to merge?

it's ok

**Pull request merged**